### ISSUE TITLE
Support BIMServer 1.5.182

### ISSUE THREAD

**Issue by dimone-kun** [xeokit/xeokit-sdk#287]
Hi @xeolabs.

Thank you for your work!

We are using xeokit in our project along side BIMServer. Starting from version 0.9.6 there are mesh loading errors on loading models from BIMServer.

```
[ERROR] [Component 'Haus']: PerformanceModel already has a Mesh with this ID: Haus.1640300.mesh Component.js:465
    error Component.js:465
    createMesh PerformanceModel.js:406
    createMeshInstancingGeometry BIMServerLoaderPlugin.js:331
    readObject BIMServerPerformanceGeometryLoader.js:503
    processMessage BIMServerPerformanceGeometryLoader.js:44
    process BIMServerPerformanceGeometryLoader.js:58
    onTick BIMServerLoaderPlugin.js:397
```

But mesh is not actualy loaded into scene.

P.S.
Cause xeokit is supporting only 1.5.120 version of BIMserver and we failed to install this version, we upgraded plugin to work with current last verision of bimserver (1.5.182). Also we [separated plugin](https://github.com/dimone-kun/xeokit-sdk-bimserver-plugin) from main repository and if you want we could transfer repo to you.

**Comment by TitouanGisle** [comment]
Hi ! Your separated plugin works great-ish !
I still have existing meshes errors, but I'm not stuck in an infinite loading anymore !
Thanks !

```
[ERROR] [Component 'myModel']: PerformanceModel already has a Mesh with this ID: myModel.177538924.mesh Component.js:481:24
[ERROR] [Component 'myModel']: Mesh with ID myModel.177538924.mesh already belongs to object with ID 17FgWlBgH3PASDyB6ea2Ss - ignoring this mesh
```

**Comment by dimone-kun** [comment]
HI, @TitouanGisle! Thank you! )

You could use xeokit-sdk 0.9.5 to fully load model.
We also stuck with problem of really high scaling (about 0.0001 in transform marix). We currently trying to solve it but we think it's in engine itself.

**Comment by xeolabs** [comment]
Hi @dimone-kun , @TitouanGisle , FYI I just merged dimone-kun's [separated plugin](https://github.com/dimone-kun/xeokit-sdk-bimserver-plugin) into xeokit-sdk master - let me know if master works OK for you..

**Comment by xeolabs** [comment]
Note that the xeokit-sdk BIMServer examples won't currently work though - the BIMServer they were pointing to is not running at the moment.

**Comment by xeolabs** [comment]
@TitouanGisle, @dimone-kun I've merged @dimone-kun's separated plugin into xeokit-sdk master (thanks for that).

Now i'm going to work on fixing the ID clashes.

**Comment by xeolabs** [comment]
@TitouanGisle, @dimone-kun - BIMServerLoaderPlugin in xeokit-sdk master should now work correctly for BIMServer 1.5.182, with ID clashes fixed.

**Comment by longho819** [comment]
Thank you so much for this fix.
I have tried fixed BIMServerLoader for BIMServer 1.5.182 and xeokit sdk 1.3.7. I got some incorrect faces of geometry and log in console: 
Fragment shader failed to compile

ERROR: 0:13: 'depth' : undeclared identifier

1: // Instancing geometry shadow drawing fragment shader
2: #ifdef GL_FRAGMENT_PRECISION_HIGH
3: precision highp float;
4: precision highp int;
5: #else
6: precision mediump float;
7: precision mediump int;
8: #endif
9: varying vec4 vViewPosition;
10: vec4 encodeFloat( const in float v ) {
11:   const vec4 bitShift = vec4(256 * 256 * 256, 256 * 256, 256, 1.0);
12:   const vec4 bitMask = vec4(0, 1.0 / 256.0, 1.0 / 256.0, 1.0 / 256.0);
13:   vec4 comp = fract(depth * bitShift);
14:   comp -= comp.xxyz * bitMask;
15:   return comp;
16: }
17: void main(void) {
18:     gl_FragColor = encodeFloat( gl_FragCoord.z); 
19: }

**Comment by xeolabs** [comment]
@longho819 no problem! 

That shader bug is fixed in 1.3.9 - I'll be releasing that this week.

Thanks for the geometry bug report, I'll see if I can fix that.

**Comment by xeolabs** [comment]
I'm going to release the ````BIMServerLoaderPlugin````` updates from @dimone-kun in 1.3.8, so that people have something to work with, then I aim to fix the geometry issue in 1.39.

**Comment by longho819** [comment]
> I'm going to release the `BIMServerLoaderPlugin` updates from @dimone-kun in 1.3.8, so that people have something to work with, then I aim to fix the geometry issue in 1.39.

thank you for your supporting

**Comment by longho819** [comment]
Hi @xeolabs, Do you have something about geometry issue? Visualize depend on this issue. Please help. Thank you
